syntax/hspec.vim

```diff
@@ -3,7 +3,7 @@
 " Maintainer:   Simon Hengel <sol@typeful.net>
 " Last Change:  2013 May 18
 
-if exists("b:current_syntax")
+if exists("b:current_syntax") && b:current_syntax == "hspec" 
```

Comment:
This is normally (part of) a boilerplate pattern for include guards that allow for syntax embedding, e.g. editing Markdown files and getting syntax highlighting for code blocks, etc. I don't see any problem with adding it, I'd just like to understand clearly what problem we're actually solving (and have a commit message that reflects it 😄 ).